### PR TITLE
Add support for C2B transaction simulation.

### PR DESCRIPTION
**Pull request by swimlappy**
Hello,

For consideration, I have added support for sending a C2B transaction simulation.  This is something that really helps during testing.  This is a fantastic package and would really like to see if this can be merged in.

Thank you!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +131,46 @@ $response = \Registrar::register(600000)
...
+```php
+use SmoDav\Mpesa\C2B\Simulate;
+
+$simulate = new Simulate($engine);
+    
+// fluent implementation
+$response = $simulate->request(10)
+    ->from(254722000000)
+    ->usingReference('f4u239fweu')
+    ->setCommand('CustomerPayBillOnline')
```
Comment (SmoDav) [inline README.md]: Add this to a constant. It's better to call `Simulate::CUSTOMER_PAYBILL_ONLINE`

**Comment by swimlappy** [comment]
Hello, have you had a chance to review this pull request?  Thanks

File: src/Mpesa/C2B/Simulate.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+     * Set the product reference number to bill the account.
+     *
+     * @param int    $reference
+     *
+     * @return $this
+     */
+    public function usingReference($reference)
+    {
+        \preg_match('/[^A-Za-z0-9]/', $reference, $matches);
+
```
Comment (SmoDav) [inline src/Mpesa/C2B/Simulate.php]: Also remove the validation as all characters are allowed.

File: src/Mpesa/C2B/Simulate.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    /**
+     * Set the unique command for this transaction type.
+     *
+     * @param string    $command
+     *
+     * @return $this
+     */
+    public function setCommand($command)
+    {
+        $valid_commands = ['CustomerPayBillOnline', 'CustomerBuyGoodsOnline'];
```
Comment (SmoDav) [inline src/Mpesa/C2B/Simulate.php]: Add these commands as constants

**Review by SmoDav — CHANGES_REQUESTED** [review]

**Comment by swimlappy** [comment]
Hello, thanks for the review.  I've added in the changes that you have requested and added to this pull request.  Please let me know if these look good.  Thank you

**Comment by swimlappy** [comment]
Hi, do the changes look ok to you?  Thanks

**Comment by swimlappy** [comment]
Hello, have you had a chance to review the additional changes?  Thanks

**Pull request merged**